Programm size leaps

2018-01-26
Mint 18.3
rustc 1.25.0-nightly (5965b7901 2018-01-19)

| example | size |
| ----- | ----- |
| blinky | 2396 |
| delay | 2460 |
| hello | 10584 |
| led | 804 |
| motor | 14222 |
| mpu9250 | 16150 |
| pwm | 12196 |
| qei | 12960 |
| reactive-blinky | 2160 |
| reactive-serial-circ | 12362 |
| serial-dma-circ | 12290 |
| serial-dma-rx | 2548 |
| serial-dma-tx | 2700 |
| serial | 12442 |

In serial example one string:
`let received = block!(rx.read()).unwrap();`
adds +9614 bytes

@japaric Not even sure how to control ThinLTO, whether I explicitly enable or disable it, I'll always get:
```
  = note: /usr/local/Cellar/gcc-arm-none-eabi-63/20170223/bin/../lib/gcc/arm-none-eabi/6.3.1/../../../../arm-none-eabi/lib/crt0.o: In function `_start':
          (.text+0xe0): undefined reference to `__libc_init_array'
          /usr/local/Cellar/gcc-arm-none-eabi-63/20170223/bin/../lib/gcc/arm-none-eabi/6.3.1/../../../../arm-none-eabi/lib/crt0.o: In function `_start':
          (.text+0xf0): undefined reference to `exit'
          collect2: error: ld returned 1 exit status
```

--- a/.cargo/config
+++ b/.cargo/config
@@ -4,6 +4,7 @@ rustflags = [
   "-C", "link-arg=-Tlink.x",
   "-C", "linker=arm-none-eabi-ld",
   "-Z", "linker-flavor=ld",
+  "-Z", "thinlto=no",
 ]
 
 [build]

